File: Assignment_12/Q5.c
#include<stdio.h>

int Count(int iNO)
{
  int iDigit = 0;
  int iCount = 0;
   
    while( iNO != 0)
    {
        iDigit = iNO % 10;

        if(iDigit < 6)
        {
            iCount++;
        }  

        iNO = iNO / 10;
    }

    return iCount;

}


int main()
{
    int iValue = 0;
    int iRet = 0;

    printf("Enter a number : \n");
    scanf("%d",&iValue);

    iRet = Count(iValue);
    
    printf("%d",iRet);
    
    return 0;
}

File: Assignment_38/Q5.java
import java.util.Scanner;

class Bitwise
{
    public int ToggleBitRange(int iNo, int iStart, int iEnd)
    {
        int iMask = 0;
        int iTempMask = 0x1;

        // Build a range mask dynamically by shifting and combining bits
        for(int i = iStart; i <= iEnd; i++)
        {
            iMask = iMask | (iTempMask << (i - 1));
        }

        int iRet = iNo ^ iMask;
        return iRet;
    }
}

public class Q5
{
    public static void main(String arg[])
    {
        Scanner sobj = new Scanner(System.in);
        Bitwise bobj = new Bitwise();

        int iValue = 0, iStart = 0, iEnd = 0, iRet = 0;

        System.out.println("Enter number : ");
        iValue = sobj.nextInt();

        System.out.println("Enter start position : ");
        iStart = sobj.nextInt();

        System.out.println("Enter end position : ");
        iEnd = sobj.nextInt();

        iRet = bobj.ToggleBitRange(iValue, iStart, iEnd);

        System.out.println("Updated no  : " + iRet);
        
        sobj.close();
    }
}
